### PR TITLE
Add 4 core static int matmul example

### PR DESCRIPTION
**Pull request by jtuyls**
@yzhang93 @MaheshRavishankar

### CODE REVIEW

File: tests/samples/matmul_fill_static_i8_i32.mlir [hunk 1]
```diff
@@ -0,0 +1,9 @@
+func.func @matmul_example(%lhs: tensor<16x256xi8>, %rhs: tensor<256x256xi8>) -> tensor<16x256xi32>
+{
+  %empty = tensor.empty() : tensor<16x256xi32>
+  %cst = arith.constant 0 : i32
+  %fill = linalg.fill ins(%cst : i32) outs(%empty : tensor<16x256xi32>) -> tensor<16x256xi32>
+  %res = linalg.matmul ins(%lhs, %rhs: tensor<16x256xi8>, tensor<256x256xi8>)
+                    outs(%fill: tensor<16x256xi32>) -> tensor<16x256xi32>
+  return %res : tensor<16x256xi32>
+}
```
Comment (yzhang93) [inline tests/samples/matmul_fill_static_i8_i32.mlir:9]: Add a new line

Comment (jtuyls) [reply tests/samples/matmul_fill_static_i8_i32.mlir]: Done

File: tests/samples/matmul_fill_spec_pack_peel.mlir [hunk 2]
```diff
@@ -4,15 +4,18 @@
...
 //   ${IREE_BUILD_DIR}/tools/iree-opt \
-//     ${IREE_AMD_AIE_DIR}/tests/samples/matmul_fill_static.mlir \
+//     ${IREE_AMD_AIE_DIR}/tests/samples/matmul_fill_static_i8_i32.mlir \
 //     --iree-hal-target-backends=amd-aie \
 //     --iree-abi-transformation-pipeline \
 //     --iree-flow-transformation-pipeline \
 //     --iree-stream-transformation-pipeline \
 //     --iree-hal-configuration-pipeline | \
 //   ${IREE_BUILD_DIR}/tools/iree-opt \
 //      --pass-pipeline='builtin.module(hal.executable(hal.executable.variant(iree-codegen-materialize-user-configs, iree-amdaie-lower-executable-target)))' \
```
Comment (yzhang93) [inline tests/samples/matmul_fill_spec_pack_peel.mlir:14]: Could you also include the change `--pass-pipeline='builtin.module(hal.executable(hal.executable.variant(iree-codegen-materialize-user-configs, iree-amdaie-lower-executable-target, fold-memref-alias-ops)))'`? I think missing `fold-memref-alias-ops` is the cause of your previous failure.

**Review by yzhang93 — APPROVED** [review]
Thanks for making the adjustments. LGTM.

**Pull request merged**